src/Node.py

```diff
@@ -305,20 +308,40 @@ def get_messages(self):
-        """
-        if self.check_quorum_threshold_for_field(ballot, 'voted'):
-            self.balloting_state['voted'][ballot.value.hash] = ballot
-            log.node.info('Quorum threshold met for \'voted\' for ballot %s. Updated state', ballot)
-
-        if self.check_quorum_threshold_for_field(ballot, 'accepted'):
-            self.balloting_state['accepted'][ballot.value.hash] = ballot
-            log.node.info('Quorum threshold met for \'accepted\' for ballot %s. Updated state.', ballot)
-
-        if self.check_quorum_threshold_for_field(ballot, 'aborted'):
-            self.balloting_state['aborted'][ballot.value.hash] = ballot
-            log.node.info('Quorum threshold met for \'aborted\' for ballot %s. Updated state.', ballot)
+        ballot_hash = ballot.value.hash
+        if 'voted' not in self.balloting_state:
+            self.balloting_state['voted'] = {}
+        if 'accepted' not in self.balloting_state:
+            self.balloting_state['accepted'] = {}
+        if 'aborted' not in self.balloting_state:
+            self.balloting_state['aborted'] = {}
+        if ballot_hash not in self.balloting_state['voted']:
+            self.balloting_state['voted'][ballot_hash] = ballot
+            log.node.info('Node %s added ballot %s to voted state', self.name, ballot)
+        elif ballot_hash in self.balloting_state['voted']:
+            existing_ballot = self.balloting_state['voted'][ballot_hash]
+            if existing_ballot.counter < ballot.counter:
+                self.balloting_state['voted'][ballot_hash] = ballot
+                log.node.info('Node %s updated ballot %s in voted state with higher counter', self.name, ballot)
+
+        if ballot_hash in self.balloting_state['aborted']:
+            aborted_ballot = self.balloting_state['aborted'][ballot_hash]
+            if aborted_ballot.counter < ballot.counter:
+                self.balloting_state['aborted'][ballot_hash] = ballot
+                log.node.info('Node %s updated ballot %s in aborted state with higher counter', self.name, ballot)
+        if self.check_Blocking_threshold(ballot.value):
+            if ballot_hash not in self.balloting_state['accepted']:
+                self.balloting_state['accepted'][ballot_hash] = ballot
+                if ballot_hash in self.balloting_state['voted']:
+                    del self.balloting_state['voted'][ballot_hash]
+                log.node.info('Node %s moved ballot %s from voted to accepted state', self.name, ballot)
+        elif self.check_quorum_threshold_for_field(ballot, "accepted"):
+            if ballot_hash not in self.balloting_state['accepted']:
+                self.balloting_state['accepted'][ballot_hash] = ballot
+                log.node.info('Node %s added ballot %s to accepted state via quorum threshold', self.name, ballot)
+        if self.check_quorum_threshold_for_field(ballot, "aborted"):
```

Comment:
I think the "check_quorum_threshold_for_field" calls will come from the receive function which will use this update state function. This function should just move ballots from 'voted' to 'accepted', or from 'accepted' to 'confirmed' in self.balloting_state